src/library/searchqueryparser.cpp

```diff
@@ -115,39 +120,54 @@ void SearchQueryParser::parseTokens(QStringList tokens,
             continue;
         }
 
+        bool exact = m_exactTextMatcher.indexIn(token) != -1;
         bool negate = token.startsWith(kNegatePrefix);
         std::unique_ptr<QueryNode> pNode;
 
         if (m_fuzzyMatcher.indexIn(token) != -1) {
             // TODO(XXX): implement this feature.
         } else if (m_textFilterMatcher.indexIn(token) != -1) {
-            QString field = m_textFilterMatcher.cap(1);
-            QString argument = getTextArgument(
-                m_textFilterMatcher.cap(2), &tokens).trimmed();
+            QString field = m_textFilterMatcher.cap(1);            
+            QString argument; 
+            if (exact) {
+                argument = getTextArgument(
+                            m_exactTextMatcher.cap(2), &tokens).trimmed();
+            } else {
+                argument = getTextArgument(
+                            m_textFilterMatcher.cap(2), &tokens).trimmed();
+            }
 
             if (!argument.isEmpty()) {
                 if (field == "crate") {
                     pNode = std::make_unique<CrateFilterNode>(
-                          &m_pTrackCollection->crates(), argument);
+                            &m_pTrackCollection->crates(), argument);
+                } else if (exact) {
+                    pNode = std::make_unique<ExactFilterNode>(
+                            m_pTrackCollection->database(),
+                            m_fieldToSqlColumns[field], argument);
                 } else {
                     pNode = std::make_unique<TextFilterNode>(
-                          m_pTrackCollection->database(), m_fieldToSqlColumns[field], argument);
+                            m_pTrackCollection->database(),
+                            m_fieldToSqlColumns[field], argument);
                 }
+            } else {
+                pNode = std::make_unique<SqlNode>(
+                        field + " IS NULL");
```

Comment:
As I previously said, this was added because the original intending here was to be able to search for tracks where the artist/genre/album... is inexistent. This is because when using the tree query helper, they appear some tracks that don't have artist but they must be shown on the tree query helper. If the user wants to see which are those tracks the fact that this query searches with `IS NULL` allows the user to find the tracks with missing artist. If we remove this line this will stop working. 

Also this was added before the exact filter was introduced so as long as there's a way to search for empty artist/genre/album we may remove it.